logstreams/src/main/java/io/camunda/zeebe/logstreams/log/LogStreamWriter.java

```diff
@@ -22,14 +39,65 @@ public interface LogStreamWriter {
   default long tryWrite(final LogAppendEntry appendEntry) {
     return tryWrite(appendEntry, LogEntryDescriptor.KEY_NULL_VALUE);
   }
+  /** {@inheritDoc} */
+  default long tryWrite(final LogAppendEntry appendEntry, final long sourcePosition) {
+    return tryWrite(Collections.singletonList(appendEntry), sourcePosition);
+  }
 
   /**
-   * Attempts to write the event to the underlying stream.
+   * Attempts to write the events to the underlying stream. This method is atomic, either all events
+   * are written, or none are.
    *
-   * @param appendEntry the entry to write
-   * @param sourcePosition a back-pointer to the record whose processing created this entry
-   * @return the event position, a negative value if fails to write the event, or 0 if the value is
-   *     empty
+   * @param appendEntries a set of entries to append; these will be appended in the order
+   * @return the last (i.e. highest) event position, a negative value if fails to write the events,
+   *     or 0 if the batch is empty
+   */
+  default long tryWrite(final LogAppendEntry... appendEntries) {
```

Comment:
:thought_balloon: This was just a convenience method, and I think it's mostly used in tests. If we don't really use it, then we could remove it, as it causes ambiguous calls with the single `tryWrite(LogAppendEntry)` I think.